source/docs/hardware-reference/talonfx/improving-performance-with-current-limits.rst

```diff
@@ -0,0 +1,215 @@
+Improving Performance with Current Limits
+=========================================
+
+Current-limiting, is the process of reducing and eliminating motor output when a given current has surpassed a limit. There are 3 types of current limits available: supply, stator, and torque. Each of these limits accomplishes different goals. This article goes over how to configure those limits, when to configure them and why current limiting is important.
+
+.. note:: By default, devices are not configured with any current limits. This is because the optimal limits depend on how the motor is integrated into the system. There are additional safety measures in place to prevent damage to the motor or motor controller under excessive load.
+
+Supply, Stator and Torque Limits
+--------------------------------
+
+It's important to understand how current limits work. When a motor is under a load, it takes an increasing amount of current to continue rotating the shaft of the motor (and by extension the mechanism). When integrating multiple motors into a system (such as an FRC robot), this increase current draw can drain batteries, reset fuses, blow breakers, or even damage the battery.
+
+There exists 3 forms of current limiting: supply, stator and torque. For all limits, the following :ref:`configs <docs/tuner/configs:tuner configs>` apply. In the below example, replace ``Supply`` with ``Stator``.
+
+.. note:: Limits for torque based control modes should be applied with ``Peak Forward Torque Current`` and ``Peak Reverse Torque Current`` instead.
+
+- ``SupplyCurrentLimit``
+- ``SupplyCurrentLimitEnable``
+
+All limits must be enabled using the appropriate enable config.
+
+Supply Limits
+^^^^^^^^^^^^^
+
+.. note:: Supply limits are not functional in torque based control modes. Use the peak torque current configs instead.
+
+When ``SupplyCurrentThreshold`` has elapsed for ``SupplyTimeThreshold`` amount of time, it will lower motor output until it's within range of the current limit (see the above section on examples of these configs).
+
+Supply current limits are useful to prevent fuses from reseting and breakers from tripping. They are also effective at preventing brownouts and reducing overall current load on the battery.
+
+Stator & Torque Limits
+^^^^^^^^^^^^^^^^^^^^^^
+
+.. important::
+
+   Stator current limits are only applicable in non-torque control modes. For example, a stator limit would have no affect if the current control mode is ``TorqueCurrentFOC``. Users utilizing torque based control modes should use ``Peak Forward Torque Current`` and ``Peak Reverse Torque Current`` configs instead, but the following documentation is still applicable.
+
+Stator current is the output current (and proportional to torque) of the motor. The stator current differs from supply current in that it's at a different voltage from the input.
```

Comment:
"Stator current is the output current of the motor and is directly proportional to torque."

Also not sure I like the second section, I'd like to expand it out to mention Power In = Power Out and how the magnitude of stator current is always >= supply current.

Reply:
Since power in = power out, V_supply * I_supply = V_stator * I_stator. Additionally, V_stator <= V_supply, which means that for the original equation to stay true, I_stator >= I_supply.

At 100% duty cycle they are equal, at every other duty cycle stator current is larger.

Reply:
I wonder if it would make sense to pull the supply current explanation further up here then?